### PR TITLE
validate type specific options

### PR DESCRIPTION
**Pull request by dbu**
@Nyholm trying to do this validation thing

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -283,4 +275,19 @@ private function addAuthenticationPluiginNode()
 
         return $node;
     }
+
+    public static function validateAuthenticationType($expected, $actual, $authName)
+    {
+        unset($actual['type']);
+        if ($expected == $actual) {
```
Comment (Nyholm) [inline DependencyInjection/Configuration.php]: Make a comment that you actually intend to do == instead of ===.

**Comment by Nyholm** [comment]
Cool. I like it. =)

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -243,24 +243,16 @@ private function addAuthenticationPluiginNode()
                     ->then(function ($config) {
                         switch ($config['type']) {
                             case 'basic':
-                                if (empty($config['username']) || empty($config['password'])) {
-                                    throw new InvalidConfigurationException('Authentication "basic" requires both "username" and "password".');
-                                }
+                                Configuration::validateAuthenticationType(['username', 'password'], array_keys($config), 'basic');
```
Comment (Nyholm) [inline DependencyInjection/Configuration.php]: Should be `self::validateAuthenticationType` right?

Comment (sagikazarmark) [inline DependencyInjection/Configuration.php]: Stupid question: since it is a closure, why can't it be `$this->validateAuthenticationType`?

Comment (dbu) [inline DependencyInjection/Configuration.php]: i was not sure with closure and scope, to be honest ;-)
will try $this->validate... can that method then even be private?

Comment (Nyholm) [inline DependencyInjection/Configuration.php]: Using $this in a closure has been supported since php 5.4. I just tried this: https://3v4l.org/gvIgS

**Comment by dbu** [comment]
fixed to use $this, and added the missing tests for the new configuration (and plugin configuration in general)

**Comment by Nyholm** [comment]
Cool. Thanks!

**Pull request merged**